### ISSUE TITLE
Bug: Windows - $'\r': command not found

### ISSUE THREAD

**Issue by Braunson** [BrainMaestro/composer-git-hooks#103]
```
Commit failed - exit code 1 received, with output: '.git/hooks/pre-commit: line 2: $'\r': command not found
committing as 
/usr/bin/env: php: No such file or directory'
```

When trying to commit via GitHub Desktop on Windows 10..

This is what my `composer.json` looks like...

```
"extra": {
    "laravel": {
        "dont-discover": []
    },
    "hooks": {
        "pre-commit": [
            "echo committing as $(git config user.name)",
            "vendor/bin/php-cs-fixer fix --using-cache=no ."
        ],
        "commit-msg": "grep -q '[A-Z]+-[0-9]+.*' $1",
        "pre-push": [
            "vendor/bin/php-cs-fixer fix --dry-run --using-cache=no .",
            "phpunit"
        ],
        "post-merge": "composer update"
    }
},

// ... some lines down

"scripts": {
    "post-install-cmd": "cghooks add --ignore-lock",
    "post-update-cmd": "cghooks update",
} 
```

**Comment by BrainMaestro** [comment]
Is this windows specific? does it work on `linux`?

**Comment by Braunson** [comment]
@BrainMaestro Yes running on Windows 10.. I'll uh have to try on Linux soon..

**Comment by Braunson** [comment]
Yeah seems Windows specific. Ran again on windows and recieved:

> Script vendor/bin/cghooks update handling the post-update-cmd event returned with error code 1

` "post-update-cmd": "vendor/bin/cghooks update",`

**Comment by Braunson** [comment]
@BrainMaestro 

> post-update-cmd: vendor/bin/cghooks update
'vendor' is not recognized as an internal or external command,
operable program or batch file.
Script vendor/bin/cghooks update handling the post-update-cmd event returned with error code 1

```
        "post-install-cmd": "vendor/bin/cghooks add --ignore-lock",
        "post-update-cmd": "vendor/bin/cghooks update",
```

Any ideas? Running it directly works, just not within composer's `post-install/update-cmd`

**Comment by BrainMaestro** [comment]
@Braunson looks like some strange behavior with the working directory that the command is run from. can you change the `post-install-cmd` to `pwd` or the windows equivalent and see what directory it's run from?

**Comment by badmansan** [comment]
I have the same issue on Win 7
```json
"extra": {
	"hooks": {
	  "pre-commit": [
		"vendor/bin/phplint"
	  ]
	}
  }
```
i try to run ```vendor/bin/cghooks pre-commit``` but get

> 'vendor' is not recognized as an internal or external command,
> operable program or batch file.

if change pre-commit section to 
```json
"pre-commit": [
		"cd",
		"vendor/bin/phplint"
	  ]
```
```vendor/bin/cghooks pre-commit``` runs only first command (```cd``` is ```pwd``` in win), and it print correct project directory

if change pre-commit section to specific windows syntaxis
```json
"pre-commit": [
		"vendor\\bin\\phplint"
	  ]
```
```vendor/bin/cghooks pre-commit``` work correctly. But in this case git can't commit:
> C:\home\test>git commit
> .git/hooks/pre-commit: line 3: vendorbinphplint: command not found

**Comment by denitsa-md** [comment]
I am experiencing the same problem on Windows 10 and I found something that _could_ be relevant. I'm trying to set up `php-cs-fixer` to run pre-commit.

My hooks look like this:

```
    "hooks": {
      "pre-commit": [
        "php-cs-fixer fix . --config .php_cs.dist --allow-risky=yes"
      ]
    }
```
_(I know this scans the entire code base, but I'm just setting it up and I haven't done the diff-ing yet)_


The relevant scripts:

```
 "scripts": {
    "post-install-cmd": "cghooks add --ignore-lock",
    "post-update-cmd": "cghooks update"
  },
```

Now the error I get when I try to commit through the Github client is:

```
Commit failed - exit code 1 received, with output: '.git/hooks/pre-commit: line 2: $'\r': command not found
/usr/bin/env: ‘php’: No such file or directory'
```

Now, I checked out the hook that gets generated in `.git/hooks/pre-commit`

It starts with `#!/bin/bash`.

If I change that to `#!/bin/sh` I can successfully do my commit.

Then I get the files fixed by php-cs-fixer as newly staged files (not sure this is excatly what I want cause I kind of want to block the commit but have to figure that out).
